### PR TITLE
Closes #436 (Consistent `generate` parameter naming scheme)

### LINKED ISSUE #436: Create consistent attack parameter naming scheme
Move away from adopting the naming scheme (arbitrarily) chosen by authors, and establish names with consistent semantics across attacks.

Proposed naming scheme:
  - "eps": the maximum perturbation applied to a clean data sample
  - "step_count": the number of steps to take in an iterative attack
  - "step_size": the size of step to take at each iteration
  - "true_label": (disjoint w/ target_label) the true one_hot label to move away from
  - "target_label": (disjoint w/ true_label) the target one_hot label to move towards
  - "clip_max": clip final adversarial example to this maximum value
  - "clip_min": clip final adversarial example to this minimum value

### PR DESCRIPTION
**Pull request by goodfeli**

### CODE REVIEW

File: cleverhans/attacks.py [hunk 1]
```diff
@@ -90,6 +90,23 @@ def generate(self, x, **kwargs):
...
+        ord: order of norm constraint
+        nb_iter: number of iterations
+        eps_iter: size of norm constraint on iteration
+        y_target: if specified, the attack is targeted.
+        y: Do not specify if y_target is specified.
+           If specified, the attack is untargeted, aims to make the output
+           class not be y.
+           If neither y_target nor y is specified, y is inferred by having
+           the model classify the input.
+      For other concepts, it's genrally a good idea to read other classes
```
Comment (npapernot) [inline cleverhans/attacks.py]: Typo

File: cleverhans/attacks.py [hunk 2]
```diff
@@ -2212,7 +2226,6 @@ def parse_params(self,
                    ord=np.inf,
                    clip_min=None,
                    clip_max=None,
-                   clip=None,
```
Comment (npapernot) [inline cleverhans/attacks.py]: do we need to deprecate this?

Comment (goodfeli) [reply cleverhans/attacks.py]: Described offline

File: cleverhans/attacks.py (hunk 1 above)
Comment (goodfeli) [reply cleverhans/attacks.py]: Fixed

**Pull request merged**